Integrations/URLHaus/URLHaus.py

```diff
@@ -0,0 +1,493 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+''' IMPORTS '''
+import os
+import traceback
+import requests
+import zipfile
+import io
+from datetime import datetime as dt
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+# Remove trailing slash to prevent wrong URL path to service
+API_URL = demisto.params()['url'].rstrip('/')
+
+# Should we use SSL
+USE_SSL = not demisto.params().get('insecure', False)
+
+# Remove proxy if not set to true in params
+if not demisto.params().get('proxy'):
+    os.environ.pop('HTTP_PROXY', None)
+    os.environ.pop('HTTPS_PROXY', None)
+    os.environ.pop('http_proxy', None)
+    os.environ.pop('https_proxy', None)
+
+THRESHOLD = int(demisto.params().get('threshold', 1))
+
+COMPROMISED_IS_MALICIOUS = demisto.params().get('compromised_is_malicious', False)
+
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Accept': 'application/json'
+}
+
+''' HELPER FUNCTIONS '''
+
+
+def http_request(method, command, data=None):
+    url = f'{API_URL}/{command}/'
+    demisto.info(f'{method} {url}')
+    res = requests.request(method,
+                           url,
+                           verify=USE_SSL,
+                           data=data,
+                           headers=HEADERS)
+
+    if res.status_code != 200:
+        raise Exception(f'Error in API call {url} [{res.status_code}] - {res.reason}')
+
+    return res
+
+
+def reformat_date(date):
+    try:
+        return dt.strptime(date.rstrip(' UTC'), '%Y-%m-%d %H:%M:%S').strftime('%Y-%m-%dT%H:%M:%S')
+    except Exception:
+        return 'Unknown'
+
+
+def extract_zipped_buffer(buffer):
+    with io.BytesIO() as bio:
+        bio.write(buffer)
+        with zipfile.ZipFile(bio) as z:
+            return z.read(z.namelist()[0])
+
+
+def query_url_information(url):
```

Comment:
Nice functions breakdown. Consider using a dictionary next time, easier to maintain in bigger integrations.

Reply:
Not sure what you mean..

Reply:
We decided to leave it for the next one.